app/javascript/schools/courses/components/curriculum_editor/CurriculumEditor__ContentBlockCreator.re

```diff
@@ -213,11 +213,63 @@ let button = (target, aboveContentBlock, send, addContentBlockCB, blockType) =>
   </label>;
 };
 
-let handleVimeoVideoUpload = (file, vimeo_video) => {
-  Js.log(vimeo_video)
-  let uploadUrl = vimeo_video##uploadUrl
+let embedUrlRegexes = [|
+  [%bs.re "/https:\\/\\/.*slideshare\\.net/"],
+  [%bs.re "/https:\\/\\/.*vimeo\\.com/"],
+  [%bs.re "/https:\\/\\/.*youtube\\.com/"],
+  [%bs.re "/https:\\/\\/.*youtu\\.be/"],
+|];
+
+let validEmbedUrl = url =>
+  Belt.Array.some(embedUrlRegexes, regex => regex->Js.Re.test_(url));
+
+let handleCreateEmbedContentBlock = 
+    (target, aboveContentBlock, url, send, addContentBlockCB) => {
+   
+  if (url |> validEmbedUrl) {
+    send(ToggleSaving);
+
+    let aboveContentBlockId =
+      aboveContentBlock |> OptionUtils.map(ContentBlock.id);
+
+    let targetId = target |> Target.id;
+
+    CreateEmbedContentBlock.make(~targetId, ~aboveContentBlockId?, ~url, ())
+    |> GraphqlQuery.sendQuery
+    |> Js.Promise.then_(result =>
+         handleGraphqlCreateResponse(
+           aboveContentBlock,
+           send,
+           addContentBlockCB,
+           result##createEmbedContentBlock##contentBlock,
+         )
+       )
+    |> Js.Promise.catch(_ => {
+         send(FailedToCreate);
+         Js.Promise.resolve();
+       })
+    |> ignore;
+  } else {
+    send(
+      SetError(
+        "The URL doesn't look valid. Please make sure that it starts with 'https://' and that it's one of the accepted websites.",
+      ),
+    );
+  };
+}
+
+let handleVimeoVideoUpload = (file, vimeoVideo, send, target, aboveContentBlock, addContentBlockCB) => {
+  Js.log(vimeoVideo)
+  let url = vimeoVideo##uri
+  let uploadUrl = vimeoVideo##uploadLink
+
+  JsTus.tusUpload(file, uploadUrl, (error) => {
+    send(FailToUpload);
+  }, 
 
-  tusUpload(file, ~uploadUrl)
+  (success) => {
+    handleCreateEmbedContentBlock(target, aboveContentBlock, url, send, addContentBlockCB);
+  })
```

Comment:
The order of arguments for error callback and success callback are inverted. However, this will become a non-issue when we switch to labelled arugments - see comment in `Tus.re`.